configure.ac

```diff
@@ -422,6 +422,7 @@ m4_foreach([cflag],[
         [-Wno-bool-operation],
         [-Wno-tautological-bitwise-compare],
         [-Wno-parentheses-equality],
+        [-Wno-parentheses],
```

Comment:
I don't think we should turn this off.

Reply:
Agreed, It's just a right pain to get righ with all the nested types and wanted to see the tests pass once. Will fix this before merging.

Reply:
Now fixed